concordium-sdk/src/main/java/com/concordium/sdk/transactions/UpdateContractTransaction.java

```diff
@@ -0,0 +1,82 @@
+package com.concordium.sdk.transactions;
+
+
+import com.concordium.sdk.exceptions.TransactionCreationException;
+import com.concordium.sdk.types.UInt64;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.val;
+
+import java.util.Objects;
+
+/**
+ * Construct a transaction to update a smart contract instance.
+ */
+@Getter
+public class UpdateContractTransaction extends AbstractTransaction {
+
+    private final UpdateContractPayload payload;
```

Comment:
I'm aware it's trivial for you what the fields are meant for. 
But please add a bit of documentation so future users can easily lookup what the fields mean/are used for.